### PR TITLE
Add support for ipaddress objects in "get" method

### PR DESCRIPTION
**Pull request by oschwald**
See also #48.

### CODE REVIEW

File: extension/maxminddb.c [hunk 1]
```diff
@@ -166,6 +171,106 @@ static PyObject *Reader_get(PyObject *self, PyObject *args)
...
+#if PY_MAJOR_VERSION >= 3
+    if (PyUnicode_Check(obj)) {
+        Py_ssize_t len;
+        const char *ipstr = PyUnicode_AsUTF8AndSize(obj, &len);
+# else
+    if (PyUnicode_Check(obj) || PyString_Check(obj)) {
+        // Although this should work on Python 3, we will hopefully delete
+        // this soon and the Python 3 version is cleaner.
+        const char *ipstr = PyString_AsString(obj);
+        Py_ssize_t len = strlen(ipstr);
```
Comment (horgh) [inline extension/maxminddb.c]: Would `PyString_Size` or something be more correct? Just thinking whether it is talking about bytes or not, and whether there could be internal NULs. But I don't know!

File: extension/maxminddb.c [hunk 2]
```diff
@@ -166,6 +171,106 @@ static PyObject *Reader_get(PyObject *self, PyObject *args)
...
+
+        struct addrinfo *addresses = NULL;
+        int gai_status = getaddrinfo(ipstr, NULL, &hints, &addresses);
+        if (gai_status) {
+            PyErr_Format(PyExc_ValueError,
+                         "'%s' does not appear to be an IPv4 or IPv6 address.",
+                         ipstr);
+            return 0;
+        }
+        memcpy(ip_address, addresses->ai_addr, addresses->ai_addrlen);
```
Comment (horgh) [inline extension/maxminddb.c]: I wonder if we should check if addresses is null. I don't know how that would be possible though.

File: extension/maxminddb.c [hunk 3]
```diff
@@ -166,6 +171,106 @@ static PyObject *Reader_get(PyObject *self, PyObject *args)
...
+        PyErr_SetString(PyExc_ValueError,
+                        "argument 1 returned an unexpected packed length for address");
+        Py_DECREF(packed);
+        return 0;
+    }
+}
+
+static char *format_sockaddr(struct sockaddr *sa)
+{
+    char *ip = calloc(INET6_ADDRSTRLEN, sizeof(char));
```
Comment (horgh) [inline extension/maxminddb.c]: Should we check success?

File: extension/maxminddb.c [hunk 4]
```diff
@@ -166,6 +171,106 @@ static PyObject *Reader_get(PyObject *self, PyObject *args)
...
+    char *addr;
+    if (sa->sa_family == AF_INET) {
+        struct sockaddr_in *sin = (struct sockaddr_in *)sa;
+        addr = (char *)&sin->sin_addr;
+    } else {
+        struct sockaddr_in6 *sin = (struct sockaddr_in6 *)sa;
+        addr = (char *)&sin->sin6_addr;
+    }
+
+    inet_ntop(sa->sa_family, addr, ip, INET6_ADDRSTRLEN);
```
Comment (horgh) [inline extension/maxminddb.c]: Should we check success?

File: extension/maxminddb.c (hunk 1 above)
Comment (oschwald) [reply extension/maxminddb.c]: I thought I tried it and it didn't work for some reason, but it seems to work now. I was trying to fix Python 2 support before my self-imposed deadline on Friday though so I was a bit rushed.

File: extension/maxminddb.c (hunk 2 above)
Comment (oschwald) [reply extension/maxminddb.c]: Hmm, I think `getaddrinfo` would have to be out of spec for that to happen, but I suppose it doesn't hurt.

**Pull request merged**